server/src/test/java/org/opensearch/action/admin/cluster/node/tasks/TaskTests.java

```diff
@@ -236,4 +238,28 @@ public void testTaskResourceStats() {
             // pass
         }
     }
+
+    public void testAddQueryGroupHeadersTo() {
+        ThreadPool threadPool = new TestThreadPool(getClass().getName());
+        try {
+            Task task = new Task(
+                randomLong(),
+                "transport",
+                SearchAction.NAME,
+                "description",
+                new TaskId(randomLong() + ":" + randomLong()),
+                Collections.emptyMap()
+            );
+
+            threadPool.getThreadContext().putHeader("queryGroupId", "afakgkagj09532059");
```

Comment:
Can you also add test covering scenario when the queryGroupId is absent from ThreadContext? Something that can happen during version upgrade, and rolling deployment